src/context/AuthContext.tsx

```diff
@@ -0,0 +1,57 @@
+import type { ReactNode} from "react";
+import { createContext, useContext, useEffect, useState } from "react"
+import { useNavigate } from "react-router-dom"
+
+interface AuthContextType {
+    authToken: string | null;
+    login: (token: string) => void;
+    logout: () => void;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined)
+
+interface AuthProviderProps {
+    children: ReactNode
+}
+
+const AuthProvider = ({ children }: AuthProviderProps) => {
+
+    const [authToken, setAuthToken] = useState<string | null>(null)
```

Comment:
```suggestion
    const [authToken, setAuthToken] = useState<string | null>(() => {
        const token = sessionStorage.getItem('authToken')

        if (token) {
            setAuthToken(token)
        }
    });
```

22~28번 라인은 useEffect를 활용해서 mount 될 때 state를 수정하고 있네요. 이러한 패턴은 보통 useState에 함수를 전달하는 방식으로 해결할 수 있어요. 리렌더링을 감소시키는 효과도 있고 초기화 코드를 상태 선언 위치에서 볼 수 있다는 장점도 있습니다. Server-side Rendering을 적용할 때는 `typeof window`처럼 window 객체의 존재 여부를 체크하고 브라우저 환경에서만 sessionStorage가 동작하도록 해야겠지만 지금은 Client-side Rendering만 적용하니 충분해보여요.

참고자료
- [useState(initiaiValue)](https://ko.react.dev/reference/react/useState#setstate)

> 함수를 initialState로 전달하면 이를 초기화 함수로 취급합니다. 이 함수는 순수해야 하고 인수를 받지 않아야 하며 반드시 어떤 값을 반환해야 합니다. React는 컴포넌트를 초기화할 때 초기화 함수를 호출하고, 그 반환값을 초기 state로 저장합니다. 아래 예시를 참고하세요.